File: investsys_back/src/main/java/com/citi/investSys/service/InvestmentService.java
package com.citi.investSys.service;

import com.citi.investSys.model.Investment;

import java.util.List;

public interface InvestmentService {
    String addInvestment(Investment investment);

    String removeInvestment(String investmentID );

    String updateInvestment(Investment investment);

    Investment getInvestment(String investmentID);

    List<Investment> getAllInvestmentsOfCustomers(String customerID);

    List<Investment> getAllInvestments();
}

File: investsys_front/src/services/investmentService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/investsys/investment'; 

const getAllInvestments = () => {
    return axios.get(API_URL);
};

const updateInvestment = (investment) =>{
    return axios.put(API_URL,investment);
}

const addInvestment = (investment) => {
    return axios.post(API_URL, investment);
};

const getInvestment = (investmentID) => {
    return axios.get(`${API_URL}/${investmentID}`);
};

const deleteInvestment = (investmentID) => {
    return axios.delete(`${API_URL}/${investmentID}`);
};

const getInvestmentsOfCustomer = (customerID) => {
    return axios.get(`${API_URL}/customer/${customerID}`);
};

export default {
    getAllInvestments,
    addInvestment,
    deleteInvestment,
    getInvestment,
    updateInvestment,
    getInvestmentsOfCustomer
};
